tuf/api/metadata.py

```diff
@@ -655,13 +717,7 @@ def update(
         hashes: Optional[Dict[str, Any]] = None,
     ) -> None:
         """Assigns passed info about snapshot metadata to meta dict."""
-        self.meta["snapshot.json"] = {"version": version}
-
-        if length is not None:
-            self.meta["snapshot.json"]["length"] = length
-
-        if hashes is not None:
-            self.meta["snapshot.json"]["hashes"] = hashes
+        self.meta["snapshot.json"] = MetaFile(version, length, hashes)
```

Comment:
If we have this function it should take a Metafile as argument, IMO.

some other opinions on update():
* if we keep it a function, it should be renamed: "update" tells me nothing
* on the other hand I kind of think this should be a property instead, maybe "snapshot_meta" so you could do things like :
`version = timestamp.signed.snapshot_meta.version
`
or 
`timestamp.signed.snapshot_meta = MetaFile(ver, len, hashes)`

Reply:
there may be an argument _against_ using helper properties like the one I handwaved: that it does not separate the attributes-that-map-the-file-format and the helpers enough (and that functions like `update()` are better in that regard). 

I think that's an issue we should consider but it does not outweigh the advantages (such as having a single read-and-write property with a descriptive name).

Reply:
About the `update()` naming there is an issue already opened capturing this #1230.

I think we should restrain ourselves from changing the API here (like changing the argument to `MetaFile` or adding properties), but focus more on the changes included in the pr. 

Instead, those could be discussed in the issue I referenced above and the issue area of focus could widen to discuss the `update()` functions design as a whole.

Reply:
I'm fine with not re-designing update here.

I disagree with leaving the argument as a dict. Why are we going through the trouble of adding objects and then not using them in the API?

Reply:
Leaving update() re-design for later is fine.

I see no reason not to make the argument a MetaFile.

Reply:
Okay, will change the argument type for the `update` functions, but won't rename it or add additional attributes as you suggested above.